File: java/DoublyLinkedListNode.java
public class DoublyLinkedListNode<T>{
    private DoublyLinkedListNode<T> next;
    private DoublyLinkedListNode<T> previous;
    private T value;
    
    public DoublyLinkedListNode(T value){
        this.value = value;
    }
    
    public T getValue(){
        return value;
    }
    
    public void setValue(T value){
        this.value = value;
    }
    
    public DoublyLinkedListNode<T> getNext(){
        return next;
    }
    
    public void setNext(DoublyLinkedListNode<T> next){
        this.next = next;
    }
    
    public DoublyLinkedListNode<T> getPrevious(){
        return previous;
    }
    
    public void setPrevious(DoublyLinkedListNode<T> previous){
        this.previous = previous;
    }
}

File: js/DoublyLinkedListNode.js
class DoublyLinkedListNode {
    #value;
    #next;
    #previous;

    constructor(value) {
        this.#value = value;
    }
    
    getValue(){
        return this.#value;
    }
    
    setValue(value){
        this.#value = value;
    }
    
    getNext(){
        return this.#next;
    }
    
    setNext(next){
        this.#next = next;
    }
    
    getPrev(){
        return this.#previous;
    }
    
    setPrev(previous){
        this.#previous = previous;
    }
}

module.exports = DoublyLinkedListNode;
